File: Updater/src/main/java/jk_5/nailed/updater/DownloadMonitor.java
package jk_5.nailed.updater;

import javax.swing.*;

/**
 * No description given
 *
 * @author jk-5
 */
@SuppressWarnings("unused")
public class DownloadMonitor {

    private final ProgressMonitor monitor;

    public DownloadMonitor(){
        monitor = new ProgressMonitor(null, "Nailed-Updater", "Checking for updates...", 0, 1);
        monitor.setMillisToPopup(0);
        monitor.setMillisToDecideToPopup(0);
    }

    public void setMaximum(int max){
        monitor.setMaximum(max);
    }

    public void setNote(String note){
        monitor.setNote(note);
    }

    public void setProgress(int progress){
        monitor.setProgress(progress);
    }

    public void close(){
        monitor.close();
    }
}

File: Updater/src/main/scala/jk_5/nailed/updater/DownloadMonitor.scala
package jk_5.nailed.updater

import javax.swing.ProgressMonitor

/**
 * No description given
 *
 * @author jk-5
 */
object DownloadMonitor {
  private val monitor = new ProgressMonitor(null, "Nailed-Updater", "Checking for updates...", 0, 1)
  this.monitor.setMillisToPopup(0)
  this.monitor.setMillisToDecideToPopup(0)

  def setMaximum(max: Int) = this.monitor.setMaximum(max)
  def setNote(note: String) = this.monitor.setNote(note)
  def setProgress(progress: Int) = this.monitor.setProgress(progress)
  def close() = this.monitor.close()
}
